### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by tobie** [heycam/webidl#520]
So, I'd like to adopt a process where we mandate tests for normative changes to the spec.

The problem is that most of the things that we're able to test must be done through testing actual concrete interfaces found in other specs. Which thus most often already have their own test suites.

For example, I just added a bunch of tests for the "is an array index" abstract operation in https://github.com/w3c/web-platform-tests/pull/9446, only to be advised in duplicated the tests found in https://github.com/w3c/web-platform-tests/pull/9303.

I'm not too sure what the solution is for this.

Should we track test suites via annotation in the spec? Should we not care and allow for duplicates? should be give-up mandating tests for normative changes in WebIDL?

What are your suggestions here?

Tagging a bunch of folks which might have an opinion about this: @annevk, @foolip, @tabatkins, @domenic, @TimothyGu, @marcoscaceres.

**Comment by tobie** [comment]
This is related to #268, but with a deliberate focus on testharness tests, not idlharness.

**Comment by domenic** [comment]
I think allowing for duplicates is fine. That's what we've done in the past.

I think requiring small testharness smoke tests of some representative interface is a good incremental step. I'm hesitant to require adding tests to idlharness because in the past you've expressed that would require rewriting idlharness, which is a major project. (Although, if we change something that makes existing idlharness tests wrong, we should then fix idlharness regardless.)

**Comment by tobie** [comment]
Yes. I don’t want to include idlharness in this process. Happy to file issues against it however, as we’ve been doing so far.

If folks are OK with duplicates, that WFM.

**Comment by marcoscaceres** [comment]
> If folks are OK with duplicates, that WFM.

I'm fine with duplicates, but there is often more test cases that can be generalized from any new IDL syntax.
